src/processTargets/modifiers/scopeHandlers/getScopeRelativeToPosition.ts

```diff
@@ -0,0 +1,36 @@
+import { Position, TextEditor } from "vscode";
+import { Direction } from "../../../typings/targetDescriptor.types";
+import { OutOfRangeError } from "../targetSequenceUtils";
+import { TargetScope } from "./scope.types";
+import { ScopeHandler } from "./scopeHandler.types";
+
+/**
+ * Runs the scope generator until `offset` scopes have been yielded and returns
+ * that scope, only yielding scopes that start after or equal {@link position}
+ * @param scopeHandler
+ * @param editor
+ * @param position
+ * @param offset
+ * @param direction
+ * @returns
+ */
+export default function getScopeRelativeToPosition(
+  scopeHandler: ScopeHandler,
+  editor: TextEditor,
+  position: Position,
+  offset: number,
+  direction: Direction,
+): TargetScope {
+  let scopeCount = 0;
+  for (const scope of scopeHandler.generateScopes(editor, position, direction, {
```

Comment:
Probably looks cleaner if we save the iterator in a constant first. Multiline function call inside the loop header is a bit cluttery for my taste tout